lib/facter/facts/linux/hypervisors/xen.rb

```diff
@@ -8,7 +8,7 @@ class Xen
 
         def initialize
           @log = Facter::Log.new(self)
-          @virtual = Facter::Util::Facts::Posix::VirtualDetector.new
+          @virtual = Facter::Util::Facts::Posix::VirtualDetector
```

Comment:
same for this, i'm more in favour of using Facter::Util::Facts::Posix::VirtualDetector directly where needed, instead of storing in in an extra instance variable.